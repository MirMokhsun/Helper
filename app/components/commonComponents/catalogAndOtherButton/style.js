import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#FFFFFF',
        borderRadius: 10,
        margin: 8,
        shadowOffset: { width: 0, height: 4 },
        shadowColor: 'rgba(0, 0, 0, 0.15)',
        shadowRadius: 4,
        elevation: 4,
        shadowOpacity: 1,
    },
    svgContainer: {
        flex: 4,
        flexDirection: 'column-reverse',
        alignItems: 'center'
    },
    svgContainerLong: {
        flex: 3,
        flexDirection: 'row-reverse',
        alignItems: 'center',
        paddingLeft: 15,
    },
    textContainer: {
        flex: 4
    },
    textContainerLong: {
        flex: 4,
    },
    title: {
        fontFamily: 'Roboto-Light',
        color: '#000756',
        fontSize: 16,
        textAlign: 'center',
    },
    text: {
        fontFamily: 'Roboto-Light',
        color: '#9FA0B2',
        fontSize: 11,
        textAlign: 'center',
        marginTop: 5
    },
    bottomLine: {
        height: 2,
        width: 'auto',
        marginHorizontal: 28,
    },
    flex1: {
        flex: 1
    },
    circle: {
        width: 24,
        height: 24,
        borderRadius: 12,
        backgroundColor: '#F86F6F',
        position: 'absolute',
        top: -8,
        right: -8,
        justifyContent: 'center',
        alignItems: 'center',
    },
    circleText: {
        fontFamily: 'Roboto-Light',
        color: '#FFFFFF',
        fontSize: 16,
        textAlign: 'center',
    }
});