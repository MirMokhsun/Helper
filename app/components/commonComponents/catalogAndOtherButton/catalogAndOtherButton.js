import React, { PureComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { styles } from './style';

export class CatalogAndOtherButton extends PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        const { title, svg, color, text, onPress, notifications, longWidth, id } = this.props;
        return (
            <TouchableOpacity style={styles.container} onPress={onPress} testID={`CatalogAndOtherButton${id}ID`} accessibilityLabel={`CatalogAndOtherButton${id}ID`}>
                {notifications ? (
                    <View style={styles.circle}>
                        <Text style={styles.circleText}>{notifications}</Text>
                    </View>
                ) : null}
                <View style={[styles.flex1, { flexDirection: longWidth ? 'row' : 'column' }]}>
                    <View style={longWidth ? styles.svgContainerLong : styles.svgContainer}>
                        {svg || null}
                    </View>
                    <View style={longWidth ? styles.textContainerLong : styles.textContainer}>
                        <View style={[styles.flex1, { justifyContent: longWidth ? 'center' : 'flex-start', minHeight: 10 }]}>
                            <Text style={[styles.title, { textAlign: longWidth ? 'left' : 'center', marginTop: longWidth ? 0 : 10 }]}>{title}</Text>
                        </View>
                        {text ? (
                            <View style={{ flex: 2 }}>
                                <Text style={styles.text}>{text}</Text>
                            </View>
                        ) : null}
                    </View>
                </View>
                <View style={[styles.bottomLine, { backgroundColor: color }]} />
            </TouchableOpacity>
        );
    }
}