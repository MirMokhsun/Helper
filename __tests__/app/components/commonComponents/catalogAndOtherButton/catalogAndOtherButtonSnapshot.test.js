import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CatalogAndOtherButton } from '../../../../../app/components/commonComponents/catalogAndOtherButton/catalogAndOtherButton';

it('create snapshot CatalogAndOtherButton', () => {
    const renderer = new ShallowRenderer()
    let catalogAndOtherButton = renderer.render(<CatalogAndOtherButton />);
    expect(catalogAndOtherButton).toMatchSnapshot();
});