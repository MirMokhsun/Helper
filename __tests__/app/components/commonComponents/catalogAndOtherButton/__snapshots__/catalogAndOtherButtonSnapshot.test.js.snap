// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot CatalogAndOtherButton 1`] = `
<TouchableOpacity
  accessibilityLabel="CatalogAndOtherButtonundefinedID"
  activeOpacity={0.2}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "elevation": 4,
      "flex": 1,
      "margin": 8,
      "shadowColor": "rgba(0, 0, 0, 0.15)",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 4,
    }
  }
  testID="CatalogAndOtherButtonundefinedID"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 4,
          "flexDirection": "column-reverse",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "justifyContent": "flex-start",
              "minHeight": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000756",
                "fontFamily": "Roboto-Light",
                "fontSize": 16,
                "textAlign": "center",
              },
              Object {
                "marginTop": 10,
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 2,
          "marginHorizontal": 28,
          "width": "auto",
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  />
</TouchableOpacity>
`;
